release-notes-agent/src/main/java/com/epam/release/notes/agent/service/action/ActionServiceProvider.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2021 EPAM Systems, Inc. (https://www.epam.com/)
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.epam.release.notes.agent.service.action;
+
+import com.epam.release.notes.agent.entity.action.Action;
+import com.epam.release.notes.agent.service.action.mail.TemplateNotificationService;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ActionServiceProvider {
+
+    private final JavaMailSender javaMailSender;
```

Comment:
those fields shouldn't be there, `ActionServiceProvider` doesn't need to know any information about internal staff of `ActionNotificationService`s. Actually this class should be a simple storage for all kind of `ActionNotificationService`s and be able to return specific object of specific `ActionNotificationService` by Action